streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -29,14 +30,20 @@
     private ProcessorContext context;
     private Deserializer<K> keyDeserializer;
     private Deserializer<V> valDeserializer;
+    private final TimestampExtractor timestampExtractor;
 
-    public SourceNode(String name, List<String> topics, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
+    public SourceNode(String name, List<String> topics, TimestampExtractor timestampExtractor, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
```

Comment:
Related to the question I have before: in the `toString` function at line 85, we are printing `topics:string`, and we canno tell if it is a pattern or a single topic right?